### PR TITLE
Update k8s dependency version to 1.21

### PR DESCRIPTION
**Pull request by xinyanw409**
Signed-off-by: wxinyan <wxinyan@wxinyan-a01.vmware.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This pr updates k8s dependency version to 1.21
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xing-yang** [comment]
@xinyanw409 which email address do you use for your Github id?  There's a CLA issue associated with your Github id that needs to get resolved.  Take a look of the info here: https://docs.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user

File: go.mod [hunk 1]
```diff
@@ -30,42 +30,42 @@ require (
 	google.golang.org/genproto v0.0.0-20201211151036-40ec1c210f7a // indirect
 	google.golang.org/grpc v1.34.0
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	k8s.io/api v0.20.0
-	k8s.io/apimachinery v0.20.0
-	k8s.io/client-go v1.20.0
-	k8s.io/component-base v0.20.0
+	k8s.io/api v1.21.0
```
Comment (xing-yang) [inline go.mod:33]: Please take a close look at the original line.  It was v0.20.0, so it should be changed to v0.21.0.
If it was v1.20.0, then change to v1.21.0.

This comment applies to all changes.

**Comment by xing-yang** [comment]
Looks like you didn't run the following:
```
go mod vendor
go mod tidy
```

**Comment by xing-yang** [comment]
The change looks good now.  Please resolve the CLA issue.

**Comment by xinyanw409** [comment]
Create another pr to resolve the CLA issue.

**Pull request closed without merging**